lib/identity/fernet_cookie_coder.rb

```diff
@@ -37,11 +39,34 @@ def decode(cipher)
     private
 
     def decode_with_key(cipher, key)
+      # There can be URL encoded characters
+      # from the cookies
+      cipher = CGI::unescape(cipher)
+      decode_with_latest_fernet(cipher, key) ||
+        decode_with_legacy_fernet(cipher, key)
+    end
+
+    def decode_with_legacy_fernet(cipher, key)
+      begin
```

Comment:
@jingweno You don't need a `begin` when the whole method body is within the begin block. You only need the `rescue` so that:

```ruby
def foobar
  fail
rescue
  puts "I caught the fail"
end
```